apis/v1alpha1/gateway_types.go

```diff
@@ -717,38 +717,140 @@ type ListenerStatus struct {
 // field.
 type ListenerConditionType string
 
+// ListenerConditionReason defined the set of reasons that explain
+// why a particular Listener condition type has been raised.
+type ListenerConditionReason string
+
+const (
+	// ListenerConditionConflicted indicates that the controller
+	// was unable to resolve conflicting specification requirements
+	// for this Listener. If a Listener is conflicted, its network
+	// port should not be configured on any network elements.
```

Comment:
I think we need a bit more detail here. What if a conflict arises after something has already been provisioned? Our standard practice here has been to accept one valid config in a conflicting situation based on those guiding principles for conflict resolution.

Reply:
There's actually no way to express that a port has both working Listeners and conflicted Listeners. The `ListenerStatus` is a slice that is keyed by port so if you had a working configuration on the port, then an update introduced a conflict, I don't see how that can be expressed.